### PR TITLE
Fix failing module test reliability workflow

### PR DESCRIPTION
**Pull request by robertschaedler3**
## Description

Update usage of `moduletest` in reliability workflow.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: .github/workflows/e2etest-run-modulestest.yml [hunk 1]
```diff
@@ -59,41 +59,31 @@ jobs:
...
           }
-          runNumber=1
+
+          cd build
+
           if [[ ${{ inputs.repeatForXMins }} -gt 0 ]];then
             end=`date -d "+${{inputs.repeatForXMins}} mins" +%s`
             while [[ $(date +%s) -lt $end ]]; do
               echo Performing test run $((++runNumber))
-              ./modulestest --gtest_output=xml:gtest-output/TestRecipes.xml
```
Comment (AhmedBM) [inline .github/workflows/e2etest-run-modulestest.yml:74]: is this no longer needed? No more gtest integration?

File: .github/workflows/e2etest-run-modulestest.yml [hunk 2]
```diff
@@ -59,41 +59,31 @@ jobs:
...
+              ./moduletest $recipes --bin $bin > ${{ matrix.distroName }}.log
               checkResult $?
             done
           else
-            ./modulestest --gtest_output=xml:gtest-output/TestRecipes.xml > ${{ matrix.distroName }}.log
+            ./moduletest $recipes --bin $bin > ${{ matrix.distroName }}.log
             checkResult $?
           fi
 
-      - name: Generate test report
```
Comment (AhmedBM) [inline .github/workflows/e2etest-run-modulestest.yml:82]: Are there no more reports generated for the failures?

File: .github/workflows/e2etest-run-modulestest.yml (hunk 1 above)
Comment (robertschaedler3) [reply .github/workflows/e2etest-run-modulestest.yml:74]: Yes correct

File: .github/workflows/e2etest-run-modulestest.yml (hunk 2 above)
Comment (robertschaedler3) [reply .github/workflows/e2etest-run-modulestest.yml:82]: Correct, we are only storing the test output as an artifact.

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**